toxcore/logger.h

```diff
@@ -61,6 +63,40 @@ void logger_write(
     const Logger *log, Logger_Level level, const char *file, int line, const char *func,
     const char *format, ...) GNU_PRINTF(6, 7);
 
+void logger_api_write(const Logger *log, Logger_Level level, const char *file, int line, const char *func,
+                      const char *format, va_list args) GNU_PRINTF(6, 0);
+
+
+/**
+ * ==================================================================================================
+ * toxcore public api call to logger, used by ToxAV
+ */
+#ifndef TOX_DEFINED
+#define TOX_DEFINED
+typedef struct Tox Tox;
+#endif /* TOX_DEFINED */
+
+void tox_logmsg(const Tox *tox, Logger_Level level, const char *file, int line, const char *func, const char *fmt,
+                ...) GNU_PRINTF(6, 7);
+
+#define LOGGER_API_TRACE(tox, ...)   LOGGER_API_WRITE(tox, LOGGER_LEVEL_TRACE  , __VA_ARGS__)
```

Comment:
Can toxav get its own logger and logging callback? If not in this PR, can that happen in a follow-up?

Reply:
there are some issues with that. toxav tests would need to be changed, we need new api functions and callbacks for the new logger, and clients would need to change.
how bad is it like its now?

Reply:
Clients only need to change if they want to continue logging toxav stuff. Whether anything gets logged at all is an implementation detail, so it doesn't break things. Changing toxav tests should not be a problem.